### ISSUE TITLE
Batch Payments response hasValidationErrors is false but payment not in Xero

### ISSUE THREAD

**Issue by GerardDR** [XeroAPI/Xero-NetStandard#324]
- Version 3.12.1

**Describe the bug**
We call CreateBatchPaymentAsync with summarizeErrors=false. The response shows HasValidationErrors=false, but the PaymentID is null and the payment does not show up in Xero.

A payment with identical details goes through fine using CreatePaymentAsync (single instead of a batched payment).

**Expected behavior**
If the payment fails then we need some indication that it failed and why it failed. The behaviour seems to b different for batches. _There are other issues too that contradict the specs_ - properties that the specs say are for NZ only such as (payment) Reference cause validation errors if left out. And invoice properties set when making the call are null in the response.

Cheers - Gerard

--- here's some data as seen in VS debugger

//--  object sent 
{class Payment {
  Invoice: class Invoice {
  Type: ACCREC
  Contact: class Contact {
  ContactID: eea5de8c-a430-4611-bab6-04e99174260b
  ContactNumber: 
  AccountNumber: 
  ContactStatus: ACTIVE
  Name: fGerard fRozario
  FirstName: fGerard
  LastName: fRozario
  EmailAddress: fgerard@hotmail.com.au
  SkypeUserName: 
  ContactPersons: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.ContactPerson]
  BankAccountDetails: 
  TaxNumber: 
  AccountsReceivableTaxType: 
  AccountsPayableTaxType: 
  Addresses: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.Address]
  Phones: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.Phone]
  IsSupplier: False
  IsCustomer: True
  DefaultCurrency: 0
  XeroNetworkKey: 
  SalesDefaultAccountCode: 
  PurchasesDefaultAccountCode: 
  SalesTrackingCategories: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.SalesTrackingCategory]
  PurchasesTrackingCategories: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.SalesTrackingCategory]
  TrackingCategoryName: 
  TrackingCategoryOption: 
  PaymentTerms: 
  UpdatedDateUTC: 18/02/2021 5:43:51 AM
  ContactGroups: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.ContactGroup]
  Website: 
  BrandingTheme: 
  BatchPayments: 
  Discount: 
  Balances: 
  Attachments: 
  HasAttachments: 
  ValidationErrors: 
  HasValidationErrors: False
  StatusAttributeString: 
}

  LineItems: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.LineItem]
  Date: 10/02/2021 12:00:00 AM
  DueDate: 10/02/2021 12:00:00 AM
  LineAmountTypes: Inclusive
  InvoiceNumber: 5ff674c8deb55f9fc4cca062~11
  Reference: Notifyre
  BrandingThemeID: aefae6d5-7bbe-4e2e-aadc-302cd07a0fc1
  Url: 
  CurrencyCode: AUD
  CurrencyRate: 1.000000
  Status: AUTHORISED
  SentToContact: False
  ExpectedPaymentDate: 
  PlannedPaymentDate: 
  CISDeduction: 
  CISRate: 
  SubTotal: 202.27
  TotalTax: 20.23
  Total: 222.50
  TotalDiscount: 
  InvoiceID: 5b4f1e90-5f9d-4830-8a04-b2e876f059eb
  RepeatingInvoiceID: 
  HasAttachments: 
  IsDiscounted: False
  Payments: 
  Prepayments: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.Prepayment]
  Overpayments: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.Overpayment]
  AmountDue: 222.50
  AmountPaid: 0.00
  FullyPaidOnDate: 
  AmountCredited: 
  UpdatedDateUTC: 18/02/2021 5:43:51 AM
  CreditNotes: 
  Attachments: 
  HasErrors: False
  StatusAttributeString: OK
  ValidationErrors: 
  Warnings: 
}

  CreditNote: 
  Prepayment: 
  Overpayment: 
  InvoiceNumber: 5ff674c8deb55f9fc4cca062~11
  CreditNoteNumber: 
  Account: class Account {
  Code: 090
  Name: 
  AccountID: 13918178-849a-4823-9a31-57b7eac713d7
  Type: 0
  BankAccountNumber: 
  Status: 0
  Description: 
  BankAccountType: 0
  CurrencyCode: 0
  TaxType: 
  EnablePaymentsToAccount: 
  ShowInExpenseClaims: 
  Class: 0
  SystemAccount: 0
  ReportingCode: 
  ReportingCodeName: 
  HasAttachments: 
  UpdatedDateUTC: 
  AddToWatchlist: 
  ValidationErrors: 
}

  Code: 
  Date: 10/02/2021 12:00:00 AM
  CurrencyRate: 
  Amount: 222.50
  Reference: 
  IsReconciled: True
  Status: 0
  PaymentType: 0
  UpdatedDateUTC: 
  PaymentID: 
  BatchPaymentID: 
  BankAccountNumber: 
  Particulars: 
  Details: Notifyre Payment
  HasAccount: 
  HasValidationErrors: 
  StatusAttributeString: 
  ValidationErrors: 
}
}

===================== RESPONSE ===============
================ BatchPayments[0]
{
class BatchPayment {
  Account: class Account {
  Code: 200
  Name: 
  AccountID: 00000000-0000-0000-0000-000000000000
  Type: 0
  BankAccountNumber: 
  Status: 0
  Description: 
  BankAccountType: 0
  CurrencyCode: 0
  TaxType: 
  EnablePaymentsToAccount: 
  ShowInExpenseClaims: 
  Class: 0
  SystemAccount: 0
  ReportingCode: 
  ReportingCodeName: 
  HasAttachments: 
  UpdatedDateUTC: 
  AddToWatchlist: 
  ValidationErrors: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.ValidationError]
}

  Reference: 
  Particulars: 
  Code: 
  Details: Notifyre Batch Payment
  Narrative: 
  BatchPaymentID: 
  DateString: 2021-02-18T05:44:12
  Date: 18/02/2021 5:44:12 AM
  Amount: 
  Payments: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.Payment]
  Type: 0
  Status: 0
  TotalAmount: 
  UpdatedDateUTC: 
  IsReconciled: false
}
}

================== response.BatchPayments.Payment[0]

{class Payment {
  Invoice: class Invoice {
  Type: ACCREC
  Contact: class Contact {
  ContactID: eea5de8c-a430-4611-bab6-04e99174260b
  ContactNumber: 
  AccountNumber: 
  ContactStatus: ACTIVE
  Name: fGerard fRozario
  FirstName: fGerard
  LastName: fRozario
  EmailAddress: fgerard@hotmail.com.au
  SkypeUserName: 
  ContactPersons: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.ContactPerson]
  BankAccountDetails: 
  TaxNumber: 
  AccountsReceivableTaxType: 
  AccountsPayableTaxType: 
  Addresses: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.Address]
  Phones: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.Phone]
  IsSupplier: False
  IsCustomer: True
  DefaultCurrency: 0
  XeroNetworkKey: 
  SalesDefaultAccountCode: 
  PurchasesDefaultAccountCode: 
  SalesTrackingCategories: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.SalesTrackingCategory]
  PurchasesTrackingCategories: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.SalesTrackingCategory]
  TrackingCategoryName: 
  TrackingCategoryOption: 
  PaymentTerms: 
  UpdatedDateUTC: 18/02/2021 5:43:51 AM
  ContactGroups: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.ContactGroup]
  Website: 
  BrandingTheme: 
  BatchPayments: 
  Discount: 
  Balances: 
  Attachments: 
  HasAttachments: 
  ValidationErrors: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.ValidationError]
  HasValidationErrors: False
  StatusAttributeString: 
}

  LineItems: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.LineItem]
  Date: 10/02/2021 12:00:00 AM
  DueDate: 10/02/2021 12:00:00 AM
  LineAmountTypes: Inclusive
  InvoiceNumber: 5ff674c8deb55f9fc4cca062~11
  Reference: Notifyre
  BrandingThemeID: aefae6d5-7bbe-4e2e-aadc-302cd07a0fc1
  Url: 
  CurrencyCode: AUD
  CurrencyRate: 1.000000
  Status: AUTHORISED
  SentToContact: False
  ExpectedPaymentDate: 
  PlannedPaymentDate: 
  CISDeduction: 
  CISRate: 
  SubTotal: 202.27
  TotalTax: 20.23
  Total: 222.50
  TotalDiscount: 
  InvoiceID: 5b4f1e90-5f9d-4830-8a04-b2e876f059eb
  RepeatingInvoiceID: 
  HasAttachments: 
  IsDiscounted: False
  Payments: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.Payment]
  Prepayments: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.Prepayment]
  Overpayments: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.Overpayment]
  AmountDue: 222.50
  AmountPaid: 0.00
  FullyPaidOnDate: 
  AmountCredited: 
  UpdatedDateUTC: 18/02/2021 5:43:51 AM
  CreditNotes: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.CreditNote]
  Attachments: 
  HasErrors: False
  StatusAttributeString: OK
  ValidationErrors: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.ValidationError]
  Warnings: 
}

  CreditNote: 
  Prepayment: 
  Overpayment: 
  InvoiceNumber: 
  CreditNoteNumber: 
  Account: 
  Code: 
  Date: 
  CurrencyRate: 
  Amount: 222.50
  Reference: 
  IsReconciled: 
  Status: 0
  PaymentType: 0
  UpdatedDateUTC: 
  PaymentID: 
  BatchPaymentID: 
  BankAccountNumber: 
  Particulars: 
  Details: Notifyre Payment
  HasAccount: 
  HasValidationErrors: 
  StatusAttributeString: 
  ValidationErrors: System.Collections.Generic.List`1[Xero.NetStandard.OAuth2.Model.Accounting.ValidationError]
}
}

**Comment by jenksguo** [comment]
Hi @GerardDR, thanks for raising this issue. We will test this scenario and get back to you on this.

**Comment by jenksguo** [comment]
Hi @GerardDR, we have found some issue with our SDK which doesn't really exposes the validationError array properly in BatchPayments object. 

When _summerisedError = false_ is used, our API will always return an 200 and show you the error in an validationError array, the order will match the BatchPayment object in sequence. This is because, we want at least some of the objects to go through and get created on Xero. It is an acknowledgement of receiving this request, not a status of the object creation. 

e.g. for below example, I raised one BatchPayment which is wrong, the API still returned OK but it has a validation error message in the response. 

Request: 
```
{
  "BatchPayments": [
    {
      "Account": {
        "AccountID": "13918178-849a-4823-9a31-57b7eac713d7"
      },
      "Details": "Batch payment details",
      "Date": "2021-03-01T04:12:58.0751976Z",
      "Amount": 1.9999
    }
  ]
}
```

Response (returned 200 OK): 
```
{
  "Id": "45df6b02-1ca6-4e50-b939-cbfef63c2b7f",
  "Status": "OK",
  "ProviderName": "DotNetAlbert",
  "DateTimeUTC": "\/Date(1614571979688)\/",
  "BatchPayments": [
    {
      "Account": {
        "AccountID": "13918178-849a-4823-9a31-57b7eac713d7",
        "ValidationErrors": []
      },
      "Details": "Batch payment details",
      "DateString": "2021-03-01T04:12:58",
      "Date": "\/Date(1614571978075)\/",
      "IsReconciled": false,
      "StatusAttributeString": "ERROR",
      "ValidationErrors": [
        {
          "Message": "You must include 1 or more payments to be batch paid"
        }
      ]
    }
  ]
}
```

The problem with our SDK is the array is not recognised for BatchPayments. So we will fix this in the next batch release (hopefully in 2 days). 

For [Account object](https://github.com/XeroAPI/Xero-NetStandard/blob/master/Xero.NetStandard.OAuth2/Model/Accounting/Account.cs#L404), we actually have this in place. 

For now, you will need to get the whole Json response to be able to see those errors. 

Post fix, you should be able to use response.BatchPayments.ValidationError to see all of the status of creations.

**Comment by GraceYBR** [comment]
Hi @GerardDR we've fixed the bug you found and have released a new version of the SDK - [3.14.4](https://www.nuget.org/packages/Xero.NetStandard.OAuth2/3.14.4). 

You can find the fix here: https://github.com/XeroAPI/Xero-NetStandard/blob/master/Xero.NetStandard.OAuth2/Model/Accounting/BatchPayment.cs#L189

![image](https://user-images.githubusercontent.com/7684236/110531169-20bd3580-8180-11eb-8b6c-8402d7595c19.png)

Thanks for your contribution to our community!
